pkg/controller/registry/resolver/cache.go

```diff
@@ -183,6 +198,12 @@ func (c *OperatorCache) Namespaced(namespaces ...string) MultiCatalogOperatorFin
 
 func (c *OperatorCache) populate(ctx context.Context, snapshot *CatalogSnapshot, registry client.Interface) {
 	defer snapshot.m.Unlock()
+	defer func() {
+		// Don't cache an errorred snapshot.
+		if snapshot.err != nil {
+			snapshot.expiry = time.Time{}
```

Comment:
does this trigger a cache-refresh hotloop?

Reply:
At worst I'd expect syncResolvingNamespace to be requeued a handful of times, but that's a flat number of retries, not a feedback cycle. Do you have a particular cycle in mind that I missed?